config/externalname.go

```diff
@@ -215,8 +215,9 @@ var ExternalNameConfigs = map[string]config.ExternalName{
 	"azurerm_ssh_public_key": config.TemplatedStringAsIdentifier("name", "/subscriptions/{{ .setup.configuration.subscription_id }}/resourceGroups/{{ .parameters.resource_group_name }}/providers/Microsoft.Compute/sshPublicKeys/{{ .external_name }}"),
 	// Disk SAS Token can be imported using the resource id
 	// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/microsoft.compute/disks/manageddisk1
-	"azurerm_managed_disk_sas_token":    config.IdentifierFromProvider,
-	"azurerm_virtual_machine_extension": config.TemplatedStringAsIdentifier("name", "{{ .parameters.virtual_machine_id }}/extensions/{{ .external_name }}"),
+	"azurerm_managed_disk_sas_token":               config.IdentifierFromProvider,
+	"azurerm_virtual_machine_extension":            config.TemplatedStringAsIdentifier("name", "{{ .parameters.virtual_machine_id }}/extensions/{{ .external_name }}"),
+	"azurerm_virtual_machine_data_disk_attachment": config.TemplatedStringAsIdentifier("name", "{{ .parameters.virtual_machine_id }}/dataDisks/{{ .external_name }}"),
```

Comment:
Since we do not have a name parameter here, I assume that this configuration will not work.

It does not seem possible to normalize the external name. Could you try the following?

```suggestion
	"azurerm_virtual_machine_data_disk_attachment": config.IdentifierFromProvider,
```

Reply:
i've changed to use intentifier from provider